src/components/Favorite.jsx

```diff
@@ -13,46 +13,35 @@ function Favorite() {
   const [currentPostLink, setCurrentPostLink] = useState("");
   const [postText, setPostText] = useState("");
   const [posts, setPosts] = useState([]);
-  const [loading, setLoading] = useState(true);
+  const [loading, setLoading] = useState(false);
 
   const rowMode = useContext(RowModeContext);
   const { enqueueSnackbar } = useSnackbar();
 
-  let savedPostsArr = JSON.parse(localStorage.getItem("posts")) || [];
-
   useEffect(() => {
-    const fetchPosts = async () => {
-      const posts = [];
-      const fetchPromises = savedPostsArr.map(async (id) => {
-        try {
-          const docRef = doc(db, "posts", id);
-          const doc = await getDoc(docRef);
-          doc?.data() && posts.push({ id: doc.id, post: doc.data() });
-        } catch (e) {
-          enqueueSnackbar("Error while getting post", {
-            variant: "error",
-          });
-        }
-      });
+    if (auth) {
+      const fetchLikedPosts = async () => {
+        setLoading(true);
+        const collections = collection(db, "posts");
+        console.log(collections);
+        const docSnap = await getDocs(collections);
 
-      try {
-        await Promise.all(fetchPromises);
-        setPosts(posts);
-        setLoading(false);
-      } catch (error) {
-        enqueueSnackbar("Error while fetching all posts", {
-          variant: "error",
+        const likedPosts = [];
+        docSnap.forEach((doc) => {
+          return likedPosts.push(doc.data());
         });
-        setLoading(false);
-      } finally {
-        setLoading(false);
-      }
-    };
 
-    if (savedPostsArr.length === 0) {
-      setLoading(false);
+        const filteredPosts = likedPosts.filter((post) =>
+          post.likecount.includes(auth.currentUser.uid)
```

Comment:
Also, favourite page is to get the posts we've saved, not liked.

Reply:
there's an array perhaps savedpostsarray in the implementation that gets the list.

Reply:
thanks for the heads up, thought it was for the liked post

Reply:
it's ok :)